content/courses/intro-to-embedded-sys-prog/chapters/low_level_programming.rst

```diff
@@ -1,38 +1,408 @@
+
+Values of type `Name` are the names of alternative machine 
+configurations supported by the implementation. `System_Name` represents 
+the current machine configuration. We've never seen any actual use of 
+this. 
+
+`Memory_Size` is an implementation-defined value that is intended to
+reflect the memory size of the configuration, in units of storage
+elements. What the value actually refers to is not specified. Is it the
+size of the address space, the amount of physical memory on the machine,
+or what? In any case, the amount of memory available to a given computer
+is neither dependent upon, nor reflected by, this constant. Consequently,
+`Memory_Size` is not useful either.
+
+Why have something defined in the language that nobody uses? In short, 
+it seemed like a good idea at the time when Ada was first defined. 
+Upward-compatibility concerns propagate these declarations forward as 
+the language evolves, just in case somebody does use them. 
 
 
 Querying Representation Choices
 -------------------------------
 
-.. todo::
+Earlier we mentioned the ability to specify a record type's layout, or a
+variable's address in memory. These are just some of the possibilities
+supported by the language in section 13. For any representation
+characteristic you can specify, you can also query that characteristic.
+That can be very handy, especially if you did not specify it and want to
+know what the compiler chose for the representation.
 
-    Complete section!
+For example, say you want to express an "overlay," in which an object of
+one type is placed at the same memory location as a distinct object of a
+distinct type, thus overlaying one object over the other. Doing so
+allows you to interact with those memory locations in more than one way:
+one way per type, specifically. This is known as "type punning" in
+computer programming. There are other was to achieve that effect as
+well, but realize that doing so circumvents the static strong typing
+used by Ada to protect us from ourselves and from others. Use it with
+care!
+
+For a concrete example, imagine a sensor providing a stream of bytes as
+input, and you know that groups of those bytes comprise a composite
+value, i.e., a value of some record type. The buffered array of inputs
+bytes would be declared in terms of a numeric, probably unsigned integer
+Byte type, whereas the composite type would, of course, be the record
+type. The clients of the sensor driver call a driver-defined procedure
+to get the latest value, passing an output argument of the record type. The
+procedure gets the next buffered bytes, converts them to the record type
+and puts the value in the parameter, maybe checks it for validity and so
+on, and then the procedure returns. An overlay would make that simple.
+We would declare an object of the record type located at the same
+address in the buffer as the starting byte representing the next
+composite value.
```

Comment:
At this point, a simple source-code example of overlays would be great. If you want, you may reuse / adapt an example from the "Ada for the Embedded C Developer" course.

Reply:
Not yet -- there's still the problem to be resolved by the next paragraphs. Then we can (and do) show the example.